marker_rustc_driver/src/conversion/marker/common.rs

```diff
@@ -70,6 +73,15 @@ impl<'ast, 'tcx> MarkerConverterInner<'ast, 'tcx> {
         unsafe { transmute(rustc_span) }
     }
 
+    #[must_use]
+    pub fn to_resugared_span_id(&self, rustc_span: rustc_span::Span) -> SpanId {
+        if let Some(source_data) = rustc_span.source_callee() {
+            self.to_span_id(source_data.call_site)
+        } else {
+            panic!("driver requested resugared id for an unsugared span: {rustc_span:#?}");
+        }
```

Comment:
```suggestion
        let source_data = rustc_span.source_callee().unwrap_or_else(|| {
          panic!("driver requested resugared id for an unsugared span: {rustc_span:#?}");
        });

        self.to_span_id(source_data.call_site)
```

Reply:
Here I don't really see the improvement of the suggesting. Having an if statement is clearer to me, and in this case, it doesn't save on indention.

Reply:
I used to think in early returns. The indentation is slightly improved, because the `self.to_span_id(...)` expression is now in closer to the left of the screen. It isn't very noticeable in this case, but in general I am used to think in this way:

1. Call a function
2. Process the error and early return or panic (consequently use `map_err(...)?` or `unwrap_or_else(|| panic!()` or `unwrap/expect`.
3. Proceed with the regular code flow on the same level of indentation.

And it helps a lot in functions where you have several functions where you process the error or panic. I suppose you didn't work with error/panic handling a lot (this is what your [comments](https://github.com/rust-marker/marker/blob/74554e7b151724c68995f91e2d6fa0207af7d8d0/cargo-marker/src/exit.rs#L1-L2) suggest, as well as some comments on other closed PRs that I saw).

From my experience using the consistent early-return approach with the minimum amount `if/else` helps keeping the code where every function call is fallible (and potentially async) much easier to read and maintain. I've written some conventions for my team, and this is one of them. You can see it [here](https://elastio.notion.site/Avoid-nesting-91ed9b81ce1f40bc9c3f268fed0445cf).

I know it's highly controversial just like the eternal argument between `to_owned/to_string`. The code in `marker_rustc_driver` is definitely not the type of code I am writing every day, so I'll leave such decisions to you. However, `cargo-marker` is much closer to my sphere, and I'd like to apply the best practices I know there in my PRs

Reply:
> And it helps a lot in functions where basically every function call is fallible and you need to propagate the error or panic. I suppose you didn't work with error/panic handling a lot (this is what your [comments](https://github.com/rust-marker/marker/blob/74554e7b151724c68995f91e2d6fa0207af7d8d0/cargo-marker/src/exit.rs#L1-L2) suggest, as well as some comments on other closed PRs that I saw).
> 
> From my experience using the consistent early-return approach with the minimum amount of `if` and especially `else` helps keeping the code where every function call is fallible (and potentially async) much easier to read and maintain. I've written some conventions for my team, and this is one of them. You can see it [here](https://elastio.notion.site/Avoid-nesting-91ed9b81ce1f40bc9c3f268fed0445cf).

I agree that this is probably the main difference. I'm used to linting code, which is basically if's all the way down. For me, the `if` statement above parses quicker in my head than the `unwrap_or_else()` one. But in `cargo-marker` I've noticed the limitations of this style.

I guess it depends on how else cases are managed. In Clippy and lints you only care about the true case. With `if-let-chains` it's basically the same indention with fewer lines of code in comparison to early returns.

I'm totally fine with you using the other style. When working on `cargo-marker` I'll probably need to revise my style, anyways.